Mathlib/AlgebraicGeometry/Morphisms/Separated.lean

```diff
@@ -128,6 +127,35 @@ lemma IsSeparated.of_comp [IsSeparated (f ≫ g)] : IsSeparated f := by
 lemma IsSeparated.comp_iff [IsSeparated g] : IsSeparated (f ≫ g) ↔ IsSeparated f :=
   ⟨fun _ ↦ .of_comp f g, fun _ ↦ inferInstance⟩
 
+instance isClosedImmersion_equalizer_ι_left {S : Scheme} {X Y : Over S} [IsSeparated Y.hom]
+    (f g : X ⟶ Y) : IsClosedImmersion (equalizer.ι f g).left := by
+  refine IsClosedImmersion.stableUnderBaseChange
+    ((Limits.isPullback_equalizer_prod f g).map (Over.forget _)).flip ?_
+  rw [← MorphismProperty.cancel_right_of_respectsIso @IsClosedImmersion _
+    (Over.prod_left_iso_pullback Y Y).hom]
+  convert (inferInstanceAs (IsClosedImmersion (pullback.diagonal Y.hom)))
+  ext1 <;> simp [Over.prod_left_iso_pullback, ← Over.comp_left]
+
+lemma ext_of_isDominant [IsReduced X] {f g : X ⟶ Y}
```

Comment:
I think this lemma deserves a docstring. Please mention something like "reduced to separated" and if you have the tag at hand, you could also use the shiny `@[stacks ...]`

Reply:
I don't think this lemma is present verbatim in stacks. It is only some lemma used in a part of a proof. I've provided a docstring for it and a stacks tag for the previous lemma.